src/overlay007/communication_club.c

```diff
@@ -0,0 +1,1731 @@
+{
+    UnkStruct_ov84_02240FA8 v0;
+
+    if (!sub_0201A7CC(&commClubMan->unk_20)) {
+        BGL_AddWindow(commClubMan->fieldSystem->unk_08, &commClubMan->unk_20, 3, param1, param2, param3, param4, 13, param5);
+    }
+
+    Window_Show(&commClubMan->unk_20, 1, 1024 - (18 + 12) - 9, 11);
+
+    v0 = param0;
+    v0.unk_00 = commClubMan->unk_64;
+    v0.unk_0C = &commClubMan->unk_20;
+
+    commClubMan->unk_5C = sub_0200112C(&v0, 0, 0, 4);
+    sub_0201A954(&commClubMan->unk_20);
+}
+
+static void CommClubMan_Init (FieldSystem * fsys)
+{
+    int v0;
+
+    GF_ASSERT((commClubMan == NULL));
+
+    commClubMan = Heap_AllocFromHeap(4, sizeof(CommClubManager));
+    MI_CpuFill8(commClubMan, 0, sizeof(CommClubManager));
+
+    commClubMan->unk_88 = 0;
+    commClubMan->fieldSystem = fsys;
+    commClubMan->unk_97 = 0;
+    commClubMan->msgLoader = MessageLoader_Init(1, 26, 353, 4);
+    commClubMan->trainerInfoPersonal = SaveData_GetTrainerInfo(FieldSystem_SaveData(commClubMan->fieldSystem));
+    commClubMan->unk_7C = TrainerInfo_New(4);
+
+    sub_0201A7A0(&commClubMan->unk_20);
+    sub_0201A7A0(&commClubMan->unk_30);
+    sub_0201A7A0(&commClubMan->unk_40);
+
+    commClubMan->unk_50 = StringFormatter_New(4);
+    commClubMan->unk_54 = StringFormatter_New(4);
+    commClubMan->unk_58 = StringFormatter_New(4);
+    commClubMan->unk_98 = 0;
+
+    for (v0 = 0; v0 < (7 + 1); v0++) {
+        commClubMan->unk_80[v0] = 0;
+    }
+
+    for (v0 = 0; v0 < 8; v0++) {
+        commClubMan->unk_00[v0] = Strbuf_Init((70 * 2), 4);
+    }
+}
+
+static void CommClubMan_Delete (void)
+{
+    int v0;
+
+    for (v0 = 0; v0 < 8; v0++) {
+        Strbuf_Free(commClubMan->unk_00[v0]);
+    }
+
+    Heap_FreeToHeap(commClubMan->unk_7C);
+
+    if (commClubMan->unk_50) {
+        sub_0200B3F0(commClubMan->unk_50);
+        commClubMan->unk_50 = NULL;
+    }
+
+    if (commClubMan->unk_54) {
+        sub_0200B3F0(commClubMan->unk_54);
+        commClubMan->unk_54 = NULL;
+    }
+
+    if (commClubMan->unk_58) {
+        sub_0200B3F0(commClubMan->unk_58);
+        commClubMan->unk_58 = NULL;
+    }
+
+    MessageLoader_Free(commClubMan->msgLoader);
+
+    if (commClubMan->unk_40.unk_0C != NULL) {
+        BGL_DeleteWindow(&commClubMan->unk_40);
+    }
+
+    Heap_FreeToHeap(commClubMan);
+    commClubMan = NULL;
+}
+
+static void CommClubMan_StartBattleClient (CommClubManager * man)
+{
+    FieldCommMan_StartBattleClient(man->fieldSystem, man->unk_91, CommClubMan_Regulation());
+}
+
+static const UnkStruct_ov84_02240FA8 Unk_ov7_0224ED34 = {
+    NULL,
+    ov7_02249C44,
+    ov7_02249C64,
+    NULL,
+    0x10,
+    0x5,
+    0x0,
+    0x8,
+    0x0,
+    0x0,
+    0x1,
+    0xF,
+    0x2,
+    0x0,
+    0x10,
+    0x1,
+    0x0,
+    0x0,
+    NULL
+};
+
+static void ov7_02249C44 (UnkStruct_0200112C * param0, u32 param1, u8 param2)
+{
+    commClubMan->unk_98 = 1;
+
+    if (param2 == 0) {
+        Sound_PlayEffect(1500);
```

Comment:
Const from `include/constants/sdat.h` (and rename it, if you have a good name for it atm)